### ISSUE TITLE
Switch to frame is not working in mobile safari(IOS).

### ISSUE THREAD

**Issue by ayosuva** [appium/appium#5832]
getting this error while switching frame 
driver.switchTo().frame(driver.findElement(By.id("FrameID")));

org.openqa.selenium.WebDriverException: 'undefined' is not an object (evaluating 'c.K.getElementById') (WARNING: The server did not provide any stacktrace information)

**Comment by ayosuva** [comment]
any update on this?

**Comment by SrinivasanTarget** [comment]
@ayosuva Can you try the same in Appium 1.5 and if you still face any issues update the logs in form of gist here.

**Comment by lunchboxdc** [comment]
I've encountered this exact issue and am experiencing it on Appium v1.5.

Here is the appium output from my call to driver.switchTo().frame()
https://gist.github.com/lunchboxdc/a2b02543c0cc839146b8

And here is the output from the call to find the element inside the iframe just after switching to it.
https://gist.github.com/lunchboxdc/a2bcf61c8d57dd4f8e27

This appears to be a duplicate of https://github.com/appium/appium/issues/5688 and https://github.com/appium/appium/issues/5560

I'm currently entirely blocked on this.

**Comment by imurchie** [comment]
Investigating. Thanks for the logs.

**Comment by srihariinnamuri** [comment]
@lunchboxdc this is the issue with real devices, we can switch to Iframe's in emulators, even I don't have any issues with Emulators. The problem is with real devices like Iphone6, 6 plus, 5 etc.. where you need to cross the webkit proxy to execute the scripts on real devices. 

We had same issue @ Nike, finally we end up with finding an work around. it's because Appium is not supporting Iframes.

**Comment by srihariinnamuri** [comment]
Please see the blow link for details
https://github.com/appium/appium/issues/2003

**Comment by ayosuva** [comment]
@srihariinnamuri ,if you have any work around for this please let me know with a sample code.

**Comment by srihariinnamuri** [comment]
every Iframe will link to the url. get the Iframe url into a variable and access it by using driver.get("url").
That means you have to override the current url with iframe url, so that you can maintain the previous data with cookies. Then fill the iframe form and go to the next screen.

This solution may not work for some of the application, but it will definatly  work for ecommerce application.

**Comment by bjharo** [comment]
I am also encountering this issue.  Our application has fields for payment information in an iframe.  Switching to that frame and filling out the fields works fine in other browsers, but does not work when I use appium to drive mobile safari on the iOS simulator.  I am using Appium 1.5, Xcode 7.2.1, and iOS 9.2.

Here's an excerpt from the Appium log:

`[HTTP] --> POST /wd/hub/session/9d707629-7151-4021-8f75-534e72840246/frame
[MJSONWP] Calling AppiumDriver.setFrame() with args: [{"ELEMENT":"5021","element-6066-11e4-a52e-4f735466cecf":"5021"},"9d707629-7151-4021-8f75-534e72840246"]
[debug] [iOS] Executing iOS command 'setFrame'
[debug] [RemoteDebugger] Executing 'get_frame_window' atom in default context
[debug] [RemoteDebugger] Sending javascript command (function(){return function...
[debug] [RemoteDebugger] Sending '_rpc_forwardSocketData:' message to remote debugger
[debug] [RemoteDebugger] Received applicationSentData response
[debug] [RemoteDebugger] Found data handler for response
[debug] [RemoteDebugger] Received data response from socket send: '"{\"status\":0,\"value\":{\...'
[debug] [RemoteDebugger] Original command: sendJSCommand
[debug] [RemoteDebugger] Received result for atom 'get_frame_window' execution: {"WINDOW":":wdc:1459176574104"}
[debug] [iOS] Entering new web frame: ':wdc:1459176574104'
[MJSONWP] Responding to client with driver.setFrame() result: null
[HTTP] <-- POST /wd/hub/session/9d707629-7151-4021-8f75-534e72840246/frame 200 507 ms - 76
[HTTP] --> POST /wd/hub/session/9d707629-7151-4021-8f75-534e72840246/element
[MJSONWP] Calling AppiumDriver.findElement() with args: ["id","spreedly-name","9d707629-7151-4021-8f75-534e72840246"]
[debug] [iOS] Executing iOS command 'findElement'
[debug] [BaseDriver] Waiting up to 0 ms for condition
[debug] [RemoteDebugger] Wrapping script for frame ':wdc:1459176574104'
[debug] [RemoteDebugger] Sending javascript command (function (window) { var do...
[debug] [RemoteDebugger] Sending '_rpc_forwardSocketData:' message to remote debugger
[debug] [RemoteDebugger] Received applicationSentData response
[debug] [RemoteDebugger] Found data handler for response
[debug] [RemoteDebugger] Received data response from socket send: '"{\"status\":13,\"value\":{...'
[debug] [RemoteDebugger] Original command: sendJSCommand
[debug] [iOS] Error received while executing atom: undefined is not an object (evaluating 'c.a.getElementById') (status: 13)`

**Comment by imurchie** [comment]
This appears to be an issue with Selenium atoms that get injected into the page to do the finding. Can you try with `findElements` (getting all, instead of one) and just using the first one?

It all works on my machine so I'm trying to get things to break. Since I've only gotten snippets of logs, is it possible to give me the url of a page that exhibits this? Thanks!

**Comment by bjharo** [comment]
@imurchie I tried findElements and got the same result.  As soon as I switch context to the iframe, I get an error whenever I try to find an element.  I can give you the url of page where I see this.  Is there a way I can message you directly?

**Comment by imurchie** [comment]
Yes. You can email it to isaac @ saucelabs.com

**Comment by MadhusaiY** [comment]
We have an issue in testing with iPhone-6 (IOS-9.2).
Issue is with the iFrame related to payment details. Our program is in Java and it is not able to select Card Type, Name on Card and other card related details.
We checked our program in desktop version of site and it is running fine on every browser but its not working for iPhone-6 on Safari browser.
We are using below code for this:

final WebElement  delFrame =CheckoutPayment.delegoFrame(driver);
driver.switchTo().frame(delFrame);
new Select(CheckoutPayment.cardType(driver)).selectByVisibleText(CUST_CCINS);
CheckoutPayment.cardName(driver).clear();
CheckoutPayment.cardName(driver).sendKeys(CUST_CCNAME);
CheckoutPayment.cardNum(driver).sendKeys(CUST_CCNUM);
CheckoutPayment.cardExpDtMnth(driver).clear();
CheckoutPayment.cardExpDtMnth(driver).sendKeys(CUST_EXPDATEMONTH);
CheckoutPayment.cardExpDtYr(driver).clear();
CheckoutPayment.cardExpDtYr(driver).sendKeys(CUST_EXPDATEYEAR);
CheckoutPayment.cardCvv(driver).clear();
CheckoutPayment.cardCvv(driver).sendKeys(CUST_CVVAL);
CheckoutPayment.cardSubmit(driver).click();
driver.switchTo().defaultContent();

**Comment by satyasrikanth** [comment]
Same issue, 

`org.openqa.selenium.support.ui.ExpectedConditions findElement
WARNING: WebDriverException thrown by findElement(By.xpath: //body)
org.openqa.selenium.WebDriverException: **undefined is not an object (evaluating 'a.nodeType') (status: 13)** (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 749 milliseconds
Build info: version: '2.53.0', revision: '35ae25b', time: '2016-03-15 16:57:40'
System info: host: 'MANSR09-E7450', ip: '155.35.142.121', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.8.0_73'
Driver info: io.appium.java_client.ios.IOSDriver
Capabilities [{networkConnectionEnabled=false, safariOpenLinksInBackground=true, databaseEnabled=false, deviceName=iPhone 5s, platform=MAC, showIOSLog=true, nativeInstrumentsLib=true, desired={showIOSLog=true, safariOpenLinksInBackground=true, nativeInstrumentsLib=true, safariIgnoreFraudWarning=true, platformVersion=9.2, safariAllowPopups=true, browserName=safari, platformName=iOS, deviceName=iPhone 5s, autoAcceptAlerts=true}, safariIgnoreFraudWarning=true, platformVersion=9.2, safariAllowPopups=true, webStorageEnabled=false, locationContextEnabled=false, browserName=safari, takesScreenshot=true, javascriptEnabled=true, platformName=iOS, autoAcceptAlerts=true}]
Session ID: a39c0606-1da7-494c-b411-b36227a362dd`

**Comment by giripalanivel** [comment]
Hi,
In our application we have below scenarios to be tested in Safari browser in iPad 8.4. I tried with Safari 8 and Appium 1.5.3
- Need to click on a button inside an iframe to play the video
- Need to select options from dropdown element which is inside the iframe. 
The iframe is available throughout the application to host different types of content.

We are able to switch to iframe but clicking the button doesn't play the video. So far tried selenium element.click,  element.sendkeys(keys.enter) and javascript click.
Similarly trying to select dropdown option inside the iframe is also not working.

There are no errors in the appium.   
Any clue on why it is behaving odd inside the iframe?

**Comment by jlipps** [comment]
@giripalanivel sounds like you're experiencing a different issue. Check out https://github.com/appium/appium/blob/master/docs/en/advanced-concepts/cross-domain-iframes.md, maybe that will help.

**Comment by margaretwax** [comment]
same issue using ios Simulator 

> 

 [HTTP] --> POST /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/element {"using":"name","value":"Войти"}

[MJSONWP] Calling AppiumDriver.findElement() with args: ["name","Войти","5394ff3b-a...
[debug] [iOS] Executing iOS command 'findElement'
[HTTP] <-- POST /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/element 500 2 ms - 144 

[HTTP] --> GET /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/source {}

[MJSONWP] Calling AppiumDriver.getPageSource() with args: ["5394ff3b-a2f6-460b-99c9-0...
[debug] [iOS] Executing iOS command 'getPageSource'
[debug] [UIAuto] Sending command to instruments: au.mainApp().getTreeForXML()

[debug] [Instruments] [INST] 2016-10-18 08:05:04 +0000 Debug: Got new command 24 from instruments: au.mainApp().getTreeForXML()

[debug] [Instruments] [INST] 2016-10-18 08:05:04 +0000 Debug: evaluating au.mainApp().getTreeForXML()

[debug] [Instruments] [INST] 2016-10-18 08:05:04 +0000 Debug: evaluation finished

[debug] [Instruments] [INST] 2016-10-18 08:05:04 +0000 Debug: responding with:true,"visible":true,"hint":null,"path":"/0/0","x":0,"y":0,"width":375,"height":667},">":[{"UIAImage":{"@":{"name":"skyeng","label":null,"value":null,"dom":null,"enabled":true,"valid":true,"visible":true,"hint":null,"path":"/0/0/0","x":130.5,"y":27,"width":114,"height":34},">":[]}},{"UIAImage":{"@":{"name":"words","label":null,"value":null,"dom":null,"enabled":true,"valid":true,"visible":true,"hint":null,"path":"/0/0/1","x":53,"y":88,"width":269,"height":55},">":[]}},{"UIACollectionView":{"@":{"name":null,"label":null,"value":"page 1 of 3","dom":null,"enabled":true,"valid":true,"visible":true,"hint":null,"path":"/0/0/2","x":0,"y":143,"width":375,"height":391},">":[{"UIACollectionCell":{"@":{"name":"Добавляйте слова в классе — учите на смартфоне","label":null,"value":0,"dom":null,"enabled":true,"valid":true,"visible":true,"hint":null,"path":"/0/0/2/0","x":0,"y":143,"width":375,"height":391},">":[{"UIAStaticText":{"@":{"name":"Добавляйте слова в классе — учите на смартфоне","label":"Добавляйте слова в классе — учите на смартфоне","value":"Добавляйте слова в классе — учите на смартфоне","dom":null,"enabled":true,"valid":true,"visible":true,"hint":null,"path":"/0/0/2/0/0","x":73,"y":437,"width":229.5,"height":44},">":[]}}]}},{"UIACollectionCell":{"@":{"name":"Научные алгоритмы тренировок — тратьте до 3 раз меньше времени  на изучение слова","label":null,"value":0,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/0/2/1","x":0,"y":0,"width":375,"height":391},">":[{"UIAStaticText":{"@":{"name":"Научные алгоритмы тренировок — тратьте до 3 раз меньше времени  на изучение слова","label":"Научные алгоритмы тренировок — тратьте до 3 раз меньше времени  на изучение слова","value":"Научные алгоритмы тренировок — тратьте до 3 раз меньше времени  на изучение слова","dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/0/2/1/0","x":0,"y":0,"width":277.5,"height":65.5},">":[]}}]}},{"UIACollectionCell":{"@":{"name":"Единая система Skyeng — прогресс изучения слов увидите  как вы, так и ваш учитель","label":null,"value":0,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/0/2/2","x":0,"y":0,"width":375,"height":391},">":[{"UIAStaticText":{"@":{"name":"Единая система Skyeng — прогресс изучения слов увидите  как вы, так и ваш учитель","label":"Единая система Skyeng — прогресс изучения слов увидите  как вы, так и ваш учитель","value":"Единая система Skyeng — прогресс изучения слов увидите  как вы, так и ваш учитель","dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/0/2/2/0","x":0,"y":0,"width":259.5,"height":65.5},">":[]}}]}}]}},{"UIAPageIndicator":{"@":{"name":null,"label":null,"value":"page 1 of 3","dom":null,"enabled":true,"valid":true,"visible":true,"hint":null,"path":"/0/0/3","x":168.5,"y":534,"width":39,"height":42},">":[]}},{"UIAButton":{"@":{"name":"Войти","label":"Войти","value":null,"dom":null,"enabled":true,"valid":true,"visible":true,"hint":null,"path":"/0/0/4","x":24,"y":579,"width":327,"height":56},">":[]}},{"UIAToolbar":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/0/5","x":0,"y":667,"width":375,"height":44},">":[{"UIAImage":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":true,"valid":
true,"visible":false,"hint":null,"path":"/0/0/5/0","x":0,"y":666.5,"width":375,"height":0.5},">":[]}},{"UIAImage":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/0/5/1","x":0,"y":667,"width":375,"height":44},">":[]}}]}}]}},{"UIAWindow":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/1","x":0,"y":0,"width":375,"height":667},">":[]}},{"UIAWindow":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/2","x":0,"y":0,"width":375,"height":667},">":[{"UIAStatusBar":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":null,"path":"/0/2/0","x":0,"y":0,"width":375,"height":20},">":[{"UIAElement":{"@":{"name":"Swipe down with three fingers to reveal the notification center., Swipe up with three fingers to reveal the control center, Double-tap to scroll to top","label":null,"value":null,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":"Swipe down with three fingers to reveal the notification center., Swipe up with three fingers to reveal the control center, Double-tap to scroll to top","path":"/0/2/0/0","x":6,"y":0,"width":39,"height":20},">":[]}},{"UIAElement":{"@":{"name":"3 of 3 Wi-Fi bars","label":"3 of 3 Wi-Fi bars","value":"SSID","dom":null,"enabled":true,"valid":true,"visible":false,"hint":"Swipe down with three fingers to reveal the notification center., Swipe up with three fingers to reveal the control center, Double-tap to scroll to top","path":"/0/2/0/1","x":50,"y":0,"width":13,"height":20},">":[]}},{"UIAElement":{"@":{"name":"2:32 PM","label":"2:32 PM","value":null,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":"Swipe down with three fingers to reveal the notification center., Swipe up with three fingers to reveal the control center, Double-tap to scroll to top","path":"/0/2/0/2","x":165,"y":0,"width":49,"height":20},">":[]}},{"UIAElement":{"@":{"name":"-100% battery power","label":"-100% battery power","value":null,"dom":null,"enabled":true,"valid":true,"visible":false,"hint":"Swipe down with three fingers to reveal the notification center., Swipe up with three fingers to reveal the control center, Double-tap to scroll to top","path":"/0/2/0/3","x":345,"y":0,"width":25,"height":20},">":[]}}]}}]}}]}}}
[debug] [Instruments] [INST] 2016-10-18 08:05:04 +0000 Debug: Running system command #25: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js /var/folders/jp/cf_8ctsj5glfkyj8_7hrngvh0000gn/T/instruments_sock 2,{"status":0,"v...

[debug] [UIAuto] Socket data received (6075 bytes)

[debug] [UIAuto] Got result from instruments: {"status":0,"value":{"UIAApplication":{"@":{"name":"Words","label":"Words","value":null,"dom":null,"enabled":true,"valid":true,"visible":true,"hint":null,"path":"/0","x":0,"y":0,"width":375,"height":667},">":[{"UIAWindow":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":true,"valid":

[MJSONWP] Responding to client with driver.getPageSource() result: "<?xml version=\"1.0\" enco...
[HTTP] <-- GET /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/source 200 1154 ms - 7711 
[HTTP] --> GET /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/orientation {}
[MJSONWP] Calling AppiumDriver.getOrientation() with args: ["5394ff3b-a2f6-460b-99c9-0...
[debug] [iOS] Executing iOS command 'getOrientation'
[debug] [UIAuto] Sending command to instruments: au.getScreenOrientation()

[debug] [Instruments] [INST] 2016-10-18 08:05:05 +0000 Debug: Got new command 25 from instruments: au.getScreenOrientation()

[debug] [Instruments] [INST] 2016-10-18 08:05:05 +0000 Debug: evaluating au.getScreenOrientation()

[debug] [Instruments] [INST] 2016-10-18 08:05:05 +0000 Debug: evaluation finished
[debug] [Instruments] [INST] 2016-10-18 08:05:05 +0000 Debug: responding with:
[debug] [Instruments] [INST] 2016-10-18 08:05:05 +0000 Debug: Running system command #26: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js /var/folders/jp/cf_8ctsj5glfkyj8_7hrngvh0000gn/T/instruments_sock 2,{"status":0,"v...

[debug] [UIAuto] Socket data received (33 bytes)

[debug] [UIAuto] Got result from instruments: {"status":0,"value":"PORTRAIT"}
[MJSONWP] Responding to client with driver.getOrientation() result: "PORTRAIT"
[HTTP] <-- GET /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/orientation 200 1064 ms - 82 

[HTTP] --> GET /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/screenshot {}
[MJSONWP] Calling AppiumDriver.getScreenshot() with args: ["5394ff3b-a2f6-460b-99c9-0...
[debug] [iOS] Executing iOS command 'getScreenshot'
[debug] [iOS] Taking screenshot: '/var/folders/jp/cf_8ctsj5glfkyj8_7hrngvh0000gn/T/appium-instruments/Run 1/screenshota10417e3-4bc3-40c1-92b7-c669b112bf7d.png'
[debug] [UIAuto] Sending command to instruments: au.capture('screenshota10417e3-4bc3-40c1-92b7-c669b112bf7d')

[debug] [Instruments] [INST] 2016-10-18 08:05:06 +0000 Debug: Got new command 26 from instruments: au.capture('screenshota10417e3-4bc3-40c1-92b7-c669b112bf7d')

[debug] [Instruments] [INST] 2016-10-18 08:05:06 +0000 Debug: evaluating au.capture('screenshota10417e3-4bc3-40c1-92b7-c669b112bf7d')

[debug] [Instruments] [INST] 2016-10-18 08:05:06 +0000 Debug: target.captureRectOnScreenWithName("{origin:{x:0.00,y:0.00}, size:{height:667.00,width:375.00}}", UIScreen, "screenshota10417e3-4bc3-40c1-92b7-c669b112bf7d")

[debug] [UIAuto] Socket data received (25 bytes)

[debug] [UIAuto] Got result from instruments: {"status":0,"value":""}
[debug] [iOS] Waiting 10000 ms for screenshot to be generated.

[debug] [Instruments] [INST] 2016-10-18 08:05:06 +0000 Screenshot captured.

[debug] [Instruments] [INST] 2016-10-18 08:05:06 +0000 Debug: evaluation finished

[debug] [Instruments] [INST] 2016-10-18 08:05:06 +0000 Debug: responding with:
[debug] [Instruments] [INST] 2016-10-18 08:05:06 +0000 Debug: Running system command #27: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js /var/folders/jp/cf_8ctsj5glfkyj8_7hrngvh0000gn/T/instruments_sock 2,{"status":0,"v...

[debug] [UIAuto] Sending command to instruments: au.getScreenOrientation()

[debug] [Instruments] [INST] 2016-10-18 08:05:07 +0000 Debug: Got new command 27 from instruments: au.getScreenOrientation()

[debug] [Instruments] [INST] 2016-10-18 08:05:07 +0000 Debug: evaluating au.getScreenOrientation()

[debug] [Instruments] [INST] 2016-10-18 08:05:07 +0000 Debug: evaluation finished
[debug] [Instruments] [INST] 2016-10-18 08:05:07 +0000 Debug: responding with:
[debug] [Instruments] [INST] 2016-10-18 08:05:07 +0000 Debug: Running system command #28: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js /var/folders/jp/cf_8ctsj5glfkyj8_7hrngvh0000gn/T/instruments_sock 2,{"status":0,"v...

[debug] [UIAuto] Socket data received (33 bytes)

[debug] [UIAuto] Got result from instruments: {"status":0,"value":"PORTRAIT"}

[MJSONWP] Responding to client with driver.getScreenshot() result: "iVBORw0KGgoAAAANSUhEUgAAAu...
[HTTP] <-- GET /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/screenshot 200 2033 ms - 89394 
[HTTP] --> GET /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/contexts {}
[MJSONWP] Calling AppiumDriver.getContexts() with args: ["5394ff3b-a2f6-460b-99c9-0...
[debug] [iOS] Executing iOS command 'getContexts'
[debug] [iOS] Getting list of available contexts
[debug] [iOS] Retrieving contexts and views
[debug] [iOS] Selecting by url: false 
[debug] [RemoteDebugger] useNewSafari --> false
[debug] [RemoteDebugger] Debugger socket connected to ::1:27753
[debug] [RemoteDebugger] Sending connection key request
[debug] [RemoteDebugger] Sending '_rpc_reportIdentifier:' message to remote debugger
[debug] [RemoteDebugger] Receiving data from remote debugger: '{"__argument":{"WIRSimulatorBuildKey":"13E230","WIRSimulatorProductVersionKey":"9.3","WIRSimulatorNameKey":"iPhone"},"__selector":"_rpc_reportSetup:"}'
[debug] [RemoteDebugger] Received response from socket send: '["iPhone","13E230","9.3"]'
[debug] [RemoteDebugger] Sim name: iPhone
[debug] [RemoteDebugger] Sim build: 13E230
[debug] [RemoteDebugger] Sim platform version: 9.3
[debug] [RemoteDebugger] Receiving data from remote debugger: '{"__argument":{"WIRApplicationDictionaryKey":{}},"__selector":"_rpc_reportConnectedApplicationList:"}'
[debug] [RemoteDebugger] Received no apps from remote debugger. Unable to connect.
[debug] [RemoteDebugger] Connected to application
[debug] [RemoteDebugger] Selecting application
[debug] [RemoteDebugger] No applications currently connected.

[debug] [UIAuto] Sending command to instruments: au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:08 +0000 Debug: Got new command 28 from instruments: au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:08 +0000 Debug: evaluating au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:08 +0000 Debug: evaluation finished
[debug] [Instruments] [INST] 2016-10-18 08:05:08 +0000 Debug: responding with:
[debug] [Instruments] [INST] 2016-10-18 08:05:08 +0000 Debug: Running system command #29: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js /var/folders/jp/cf_8ctsj5glfkyj8_7hrngvh0000gn/T/instruments_sock 2,{"status":0,"v...

[debug] [UIAuto] Socket data received (28 bytes)

[debug] [UIAuto] Got result from instruments: {"status":0,"value":false}

[HTTP] --> GET /wd/hub/status {}

[MJSONWP] Calling AppiumDriver.getStatus() with args: []

[MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.5.3"...
[HTTP] <-- GET /wd/hub/status 200 10 ms - 83 

[debug] [UIAuto] Sending command to instruments: au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:13 +0000 Debug: Got new command 29 from instruments: au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:13 +0000 Debug: evaluating au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:13 +0000 Debug: evaluation finished
[debug] [Instruments] [INST] 2016-10-18 08:05:13 +0000 Debug: responding with:
[debug] [Instruments] [INST] 2016-10-18 08:05:13 +0000 Debug: Running system command #30: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js /var/folders/jp/cf_8ctsj5glfkyj8_7hrngvh0000gn/T/instruments_sock 2,{"status":0,"v...

[debug] [UIAuto] Socket data received (28 bytes)

[debug] [UIAuto] Got result from instruments: {"status":0,"value":false}

[debug] [UIAuto] Sending command to instruments: au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:18 +0000 Debug: Got new command 30 from instruments: au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:18 +0000 Debug: evaluating au.alertIsPresent()

[debug] [Instruments] [INST] 2016-10-18 08:05:19 +0000 Debug: evaluation finished
[debug] [Instruments] [INST] 2016-10-18 08:05:19 +0000 Debug: responding with:
[debug] [Instruments] [INST] 2016-10-18 08:05:19 +0000 Debug: Running system command #31: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js /var/folders/jp/cf_8ctsj5glfkyj8_7hrngvh0000gn/T/instruments_sock 2,{"status":0,"v...

[debug] [UIAuto] Socket data received (28 bytes)

[debug] [UIAuto] Got result from instruments: {"status":0,"value":false}

[debug] [iOS] No web frames found.

[MJSONWP] Responding to client with driver.getContexts() result: ["NATIVE_APP"]

[HTTP] <-- GET /wd/hub/session/5394ff3b-a2f6-460b-99c9-02db85a3d2a1/contexts 200 15074 ms - 86

**Comment by giripalanivel** [comment]
@jlipps
Sorry, both the iframes sources are in same domain. Basically all the contents are internal resources of the application. 
URLs will be appended with content url once we click/move to iframe.

**Comment by lawred2** [comment]
Is there a mechanism by which to stop the RemoteDebugger truncating it's debug logs 

i.e, [debug] [RemoteDebugger] Sending javascript command (function (window) { var do...

**Comment by imurchie** [comment]
No. You can update it locally and run, but the JS being sent is minified Selenium atoms, and they have never proven to be useful to be able to see.

**Comment by aluzardo** [comment]
Same issue using Appium 1.6.4-beta, iphone 6, ios version 10.2 and safari.

I am able to switch to frame but getting one error when I try to search one element.

```
WebDriverError: An unknown server-side error occurred while processing the command. Original error: Error while executing atom: undefined is not an object (evaluating 'b.querySelectorAll') (status: 13)
```

**Comment by nagarajan-ganesan** [comment]
any latest update on this issue?

**Comment by mikesalvia** [comment]
I am experiencing similar issues when switching to iframes as well! Are there any logs I can help provide?

**Comment by bhumireddy100** [comment]
I am also facing same issue. Not able to perform action on fields under iframe.

Environment:
iPad real device : version 10.2.1
Browser: Safari
Appium: 1.6.3

**Comment by ZoharLiran** [comment]
Same issue here. Environment:
iPad Air simulator, iOS 9.3

**Comment by midopa** [comment]
From #2003, which I didn't realize was closed:

I had a similar problem and worked around this issue by using a reverse proxy that routed traffic based on request paths.

I also had to get the test site to change the domain it loads the inner iframe content from. This step really depends on the kind of hooks and configuration capabilities your test environment & product have.

I forget the exact details (it was a former job and I have bad memory), but I used Nginx and had a server config that looked something like:

server {
   server_name tests.blahblah.com;
   
   # for inner iframe, as accessed from parent page
   location /inner/iframe/access/path {
      # forward requests to actual url for inner iframe content
      proxy_pass http://dfhkdfsdh.com/content
   }

   # everything else accessed on that parent page
   location / {
      # forward requests to actual url for parent page
      proxy_pass http://asdsfhgjhe.com
   }
}
Then I'd point my tests to go to tests.blahblah.com. Since both the parent and inner iframe are now in the same domain, Safari won't block access due to CORS violations. Then Appium/Webdriver has no trouble switching into that iframe.

Sorry if the details are scant (and likely wrong), my memory is bad, I've left the previous company, and I don't save my work on my personal machines.

**Comment by sashwin** [comment]
Same issue here. Our site has an iframe within which another webapp is opened. We are unable to test this due to this. It works perfectly fine with selenium.
Environment : iPad Air simulator, iOS 9.3

**Comment by mec07** [comment]
I'm having the same issue. I'm trying to use selenium to test stripe elements on sauce labs on iOS.

The error I get when I look for an element inside the iframe and I search by class name:
```      Traceback (most recent call last):
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/behave/model.py", line 1456, in run
          match.run(runner.context)
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/behave/model.py", line 1903, in run
          self.func(context, *args, **kwargs)
        File "features/steps/white_label.py", line 281, in when_i_enter_my_card_details
          card_elem = context.driver.find_element_by_class_name("CardField-input-wrapper")
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py", line 426, in find_element_by_class_name
          return self.find_element(by=By.CLASS_NAME, value=name)
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py", line 784, in find_element
          'value': value})['value']
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py", line 249, in execute
          self.error_handler.check_response(response)
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/selenium/webdriver/remote/errorhandler.py", line 193, in check_response
          raise exception_class(message, screen, stacktrace)
      selenium.common.exceptions.WebDriverException: Message: undefined is not an object (evaluating 'a.querySelectorAll')```

and the error I get when I search by xpath is:
```      Traceback (most recent call last):
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/behave/model.py", line 1456, in run
          match.run(runner.context)
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/behave/model.py", line 1903, in run
          self.func(context, *args, **kwargs)
        File "features/steps/white_label.py", line 281, in when_i_enter_my_card_details
          card_elem = context.driver.find_element_by_xpath("html/body/form/div")
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py", line 306, in find_element_by_xpath
          return self.find_element(by=By.XPATH, value=xpath)
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py", line 784, in find_element
          'value': value})['value']
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py", line 249, in execute
          self.error_handler.check_response(response)
        File "/Users/mcoury/.virtualenvs/hlin/lib/python3.6/site-packages/selenium/webdriver/remote/errorhandler.py", line 193, in check_response
          raise exception_class(message, screen, stacktrace)
      selenium.common.exceptions.WebDriverException: Message: undefined is not an object (evaluating 'a.nodeType')```

**Comment by soul3510** [comment]
Same issue with iPhone 7, 7 Plus - can't switch to i-frame at all on Safari )-:
Getting the following error: 
undefined is not an object (evaluating 'a.nodeType')
Any help will be appreciate.

**Comment by jlipps** [comment]
@soul3510 what is the url of your main webview and what is the url of the iframe src?

**Comment by jlipps** [comment]
closing due to inactivity

**Comment by svetikshark** [comment]
Having issue with swtiching to iframe on xcode simulator
xcode 8.1
appium 1.6.4.
IPhone 6 10.3
After switching to iframe context still has main window and any other operations throw an errors so that I can't switch back with WrappedDriver.SwitchTo().DefaultContent();

WrappedDriver.SwitchTo().DefaultContent()' threw an exception of type 'OpenQA.Selenium.NoSuchElementException'
    Data: {System.Collections.ListDictionaryInternal}
    HResult: -2146233088
    HelpLink: null
    InnerException: null
    Message: "An element could not be located on the page using the given search parameters."

**Comment by cesarsruiz** [comment]
@ayosuva You didn't provide the appium server log

**Comment by denslinsunitha** [comment]
I am facing same issue. While switching to iframe getting error as below.
Appium : 1.6.4
Ios: 10.3
Mobile browser: safari

logs:
org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. Original error: Error while executing atom: Blocked a frame with origin "https://ukuatcmsweb.hofssc.com" from accessing a frame with origin "https://upayment.hofssc.com". Protocols, domains, and ports must match. (status: 18) (WARNING: The server did not provide any stacktrace information)

Command duration or timeout: 16 milliseconds

Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'

System info: host: 'ChristisMacbook', ip: 'fe80:0:0:0:c0e:3cd5:32a8:401%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.6', java.version: '1.8.0_144'

Driver info: io.appium.java_client.ios.IOSDriver

Capabilities [{applicationCacheEnabled=true, networkConnectionEnabled=false, appiumVersion=1.6.4, unhandledPromptBehavior=dismiss, databaseEnabled=false, handlesAlerts=false, deviceName=iPhone 6, nativeWebTap=true, platform=MAC, ensureCleanSession=true, acceptSslCerts=true, platformVersion=10.3, webStorageEnabled=false, locationContextEnabled=false, browserName=Safari, takesScreenshot=true, javascriptEnabled=true, platformName=iOS, udid=1259D399-D2CC-4905-B613-2C5F2C4A3202, unexpectedAlertBehaviour=dismiss}]

If anybody has solution for this issue can u pls share it ?

**Comment by Gaurang033** [comment]
I am getting the same error, I am running on the **iPad Mini 4** simulators. 

Code that is giving error. 

```
public void fillCreditCardInformation(String cardNumber){
        wait.until(ExpectedConditions.frameToBeAvailableAndSwitchToIt(By.name("__privateStripeFrame3")));
        WebElement cardNumberEditbox = driver.findElement(By.name("cardnumber"));
}

```
**Webdriver:** 3.4.0 tried with 3.5.2 as well. same error. 
**Safari Driver:** 2.48

**StackTrace:**
`org.openqa.selenium.WebDriverException: undefined is not an object (evaluating 'a.querySelectorAll') (WARNING: The server did not provide any stacktrace information) Command duration or timeout: 30.48 seconds Build info: version: '3.4.0', revision: 'unknown', time: 'unknown' System info: host: 'DL0019', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.0-96-generic', java.version: '1.8.0_131' Driver info: org.openqa.selenium.remote.RemoteWebDriver Capabilities [{noReset=true, safari=true, browserstack.tunnelIdentifier=, browserstack.asyncStop=true, language=Apache-HttpClient/4.5.3 (Java/1.8.0_131), browserstack.selenium_version=3.4.0, deviceName==iPad Retina (9.1) [556DF534-C4AE-48B6-8ED3-BAF86198074A], platform=MAC, browserstack.video.disableWaterMark=true, desired={noReset=true, browserstack.tunnelIdentifier=, browserstack.asyncStop=true, language=Apache-HttpClient/4.5.3 (Java/1.8.0_131), browserstack.selenium_version=3.4.0, deviceName==iPad Retina (9.1) [556DF534-C4AE-48B6-8ED3-BAF86198074A], platform=MAC, browserstack.video.disableWaterMark=true, acceptSslCerts=false, newCommandTimeout=300.0, browser=ipad, platformVersion=9.1, acceptSslCert=false, browserName=safari, platformName=iOS, 64bit=false, browserstack.debug=true, orientation=portrait, browserstack.ie.noFlash=false, os_version=, mobile={"browser":"tablet","version":"iPad Mini 4-9.1"}, browserstack.geckodriver=0.16.0, version=, browserstack.video=true, safariIgnoreFraudWarning=true, orig_os=macelc, realMobile=false, deviceOrientation=PORTRAIT, device=iPad Retina, proxy_type=node}, acceptSslCerts=false, newCommandTimeout=300.0, browser=ipad, platformVersion=9.1, webStorageEnabled=false, acceptSslCert=false, browserName=safari, takesScreenshot=true, javascriptEnabled=true, platformName=iOS, 64bit=false, browserstack.debug=true, networkConnectionEnabled=false, orientation=portrait, browserstack.ie.noFlash=false, warnings={}, os_version=, mobile={"browser":"tablet","version":"iPad Mini 4-9.1"}, browserstack.geckodriver=0.16.0, databaseEnabled=false, version=, browserstack.video=true, safariIgnoreFraudWarning=true, orig_os=macelc, realMobile=false, locationContextEnabled=false, deviceOrientation=PORTRAIT, device=iPad Retina, proxy_type=node}] Session ID: 6bf643515813d0ccbe5fe75300ac2d8ea15a5960 *** Element info: {Using=name, value=cardnumber}`

**Comment by VijilPatodi** [comment]
I am facing the same issue on Iphone8, appium 1.7  any work around for this will help? i am using C# dot net as language. how to do reverse proxy through c# code

**Comment by jenniferneale** [comment]
I have the same issue on iPhone8, appium 1.7 with Java.

**Comment by pmohanalatha** [comment]
I am facing same issue, elements are not identified on iframes. We have a scenario where credit card payment field to be filled.  When I tried "Srihariinnamuri" workaround accessing iFrame url, the page is not displaying the fields. Is there any fix found / any plan to release in upcoming versions.
Environment: Xcode- 9.1, iOS, iPhone 7 plus, Mobile Browser: internal chrome, Appium - 1.7.1
